src/main/java/backend/resource/TurboLabel.java

```diff
@@ -4,153 +4,185 @@
 import backend.resource.serialization.SerializableLabel;
 import javafx.scene.Node;
 import javafx.scene.control.Tooltip;
+
 import org.eclipse.egit.github.core.Label;
 
+import util.Utility;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
-@SuppressWarnings("unused")
 public class TurboLabel implements Comparable<TurboLabel> {
 
+    public enum Grouping { EXCLUSIVE, NON_EXCLUSIVE, NONE }
+
+    public static final String DEFAULT_COLOR = "ffffff";
     public static final String EXCLUSIVE_DELIMITER = ".";
     public static final String NONEXCLUSIVE_DELIMITER = "-";
 
-    private void ______SERIALIZED_FIELDS______() {
-    }
-
-    private final String actualName;
-    private final String colour;
-
-    private void ______TRANSIENT_FIELDS______() {
-    }
+    private final String fullName;
+    private final String shortName;
+    private final String groupName;
+    private final Grouping grouping;
 
     private final String repoId;
 
-    private void ______CONSTRUCTORS______() {
-    }
+    @SuppressWarnings("PMD")
+    private String colour;
```

Comment:
Can this be final? It's private and only initialised in constructors. Then we don't need to suppress the warning.